azure/shared/client.js

```diff
@@ -0,0 +1,151 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2019 Datadog, Inc.
+
+var tls = require('tls');
+
+const VERSION = '0.1.0';
+
+const STRING = 'string'; // example: 'some message'
+const STRING_ARRAY = 'string-array'; // example: ['one message', 'two message', ...]
+const JSON_OBJECT = 'json-object'; // example: {"key": "value"}
+const JSON_RECORDS = 'json-records'; // example: [{"records": [{}, {}, ...]}, {"records": [{}, {}, ...]}, ...]
+const JSON_ARRAY = 'json-array'; // example: [{"key": "value"}, {"key": "value"}, ...]
+const INVALID = 'invalid';
+
+var DD_API_KEY = process.env.DD_API_KEY || '<DATADOG_API_KEY>';
+var DD_URL = process.env.DD_URL || 'functions-intake.logs.datadoghq.com';
+var DD_TAGS = process.env.DD_TAGS || '<TAG_KEY>:<TAG_VALUE>';
+var DD_PORT = process.env.DD_PORT || 10516;
+var DD_SERVICE = process.env.DD_SERVICE || 'azure';
+var DD_SOURCE = process.env.DD_SOURCE || 'azure';
+var DD_SOURCE_CATEGORY = process.env.DD_SOURCE_CATEGORY || 'azure';
+
+function getSocket(context) {
+    var socket = tls.connect({ port: DD_PORT, host: DD_URL });
+    socket.on('error', err => {
+        context.log.error(err.toString());
+        socket.end();
+    });
+
+    return socket;
+}
+
+function send(socket, record) {
+    return socket.write(DD_API_KEY + ' ' + JSON.stringify(record) + '\n');
+}
+
+function handleLogs(handler, logs, context) {
```

Comment:
I think you should create a new class `LogsHandler` that wraps the sending logic and tagging logic and just call `logsHandler.handler(logs, context)` from the outside world.
Would be nice maybe to inject a client for the sending logic and tagger for the tagging logic.